lib/automaticupgrades/version.go

```diff
@@ -73,3 +72,60 @@ func Version(ctx context.Context, baseURL string) (string, error) {
 
 	return versionString, trace.Wrap(err)
 }
+
+// getVersionURL returns the versionURL or the default stable/cloud version url.
+func getVersionURL(versionURL string) (string, error) {
+	if versionURL != "" {
+		return versionURL, nil
+	}
+	cloudStableVersionURL, err := url.JoinPath(stableCloudVersionBaseURL, stableCloudVersionPath)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	return cloudStableVersionURL, nil
+}
+
+// Critical returns true if a critical upgrade is available.
+func Critical(ctx context.Context, criticalURL string) (bool, error) {
+	criticalURL, err := getCriticalURL(criticalURL)
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, criticalURL, nil)
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
```

Comment:
Do we plan to support HTTPS proxy ?

Reply:
This topic has not been discussed for this task. Do you think we need support in this PR. Otherwise we can create an issue and implement in a future PR.